cli_tools/diagnostics/main_windows.go

```diff
@@ -0,0 +1,241 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	sys32 = `C:\Windows\System32`
+)
+
+type runner interface {
+	run() (string, error)
+}
+
+type cmd struct {
+	name           string
+	args           string
+	outputFileName string
+	// True when the command produces its own file and doesn't need one
+	// created from stdout.
+	cmdProducesFile bool
+}
+
+type wmiQuery struct {
+	class          string
+	namespace      string
+	outputFileName string
+}
+
+func (command cmd) run() (string, error) {
+	outPath := filepath.Join(tmpFolder, command.outputFileName)
+
+	cmdPath := filepath.Join(sys32, command.name)
```

Comment:
Don't do this here, use the full path when creating the cmd so that you can run files not just in sys32

Reply:
Done